src/providers/RelayProvider.ts

```diff
@@ -0,0 +1,36 @@
+import { Observable } from 'rxjs/Observable'
+
+import { TLProvider } from './TLProvider'
+
+import { fetchUrl, trimUrl, websocketStream } from '../utils'
+
+export class RelayProvider implements TLProvider {
+  public relayApiUrl: string
+  public relayWsApiUrl: string
+
+  constructor(relayApiUrl: string, relayWsApiUrl: string) {
+    this.relayApiUrl = relayApiUrl
+    this.relayWsApiUrl = relayWsApiUrl
+  }
+
+  public async fetchEndpoint<T>(
```

Comment:
What do you think of `fetch(endpoint, options)`
Do you think this has the same amount of information, or do you think `fetchEndpoint(endpoint, options)` is better?

Reply:
I also thought about naming it only `fetch` but could also be confusing as there is already a fetch api in javascript :thinking: Or maybe `fetchJson`? I am pretty indifferent here..

Reply:
I will leave it as `fetchEndpoint` for now.